src/main/java/ch/njol/skript/expressions/ExprDateAgoLater.java

```diff
@@ -45,28 +45,37 @@
                 "%timespan% (later|(from|after) [the] [date] %-date%)");
     }
 
+    @Override
+    @SuppressWarnings({"unchecked", "null"})
+    public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+        timespan = (Expression<Timespan>) exprs[0];
+        date = (Expression<Date>) exprs[1];
+        ago = matchedPattern == 0;
+        return true;
+    }
+
     @SuppressWarnings("null")
     private Expression<Timespan> timespan;
     @Nullable
     private Expression<Date> date;
 
     private boolean ago;
 
-    @SuppressWarnings("null")
-    @Nullable
     @Override
+    @Nullable
+    @SuppressWarnings("null")
     protected Date[] get(Event e) {
         Timespan timespan = this.timespan.getSingle(e);
-        Date date = this.date == null ? new Date() : this.date.getSingle(e);
-        if (timespan == null || date == null) {
+        Date date = this.date == null ? new Date() : new Date(this.date.getSingle(e).getTimestamp());
```

Comment:
You should check if getSingle returns null before calling any methods

Reply:
Thank you for letting me know, I will do that in a moment \:D